libpod/container_api.go

```diff
@@ -351,7 +353,7 @@ func (c *Container) Exec(tty, privileged bool, env, cmd []string, user, workDir
 	if err := c.save(); err != nil {
 		logrus.Errorf("Error removing exec session %s from container %s state: %v", sessionID, c.ID(), err)
 	}
-
+	c.newContainerEvent(events.Exec)
```

Comment:
This needs to go a lot further up, right after we start the exec but before we unlock and wait for it. Still needs to not be a defer.